src/db/mod.rs

```diff
@@ -55,7 +59,38 @@ where
         if let Some(max_size) = database_config.max_conns {
             builder = builder.max_size(max_size)
         }
-        Self::from_pool_builder(&database_config.url, builder)
+
+        #[cfg(feature = "sqlite")]
+        let database_url = database_config.url.as_str();
+        #[cfg(any(feature = "mysql", feature = "postgres"))]
+        let database_url = {
+            let mut url = Url::parse(database_config.url.as_str()).expect("invalid connection URL");
+            if let Some(user) = database_config.user.as_ref() {
+                if url.username().is_empty() {
+                    url.set_username(user.as_str())
+                        .expect("could not append username to the connection URL");
+                } else {
+                    panic!("conflicting usernames in database configuration");
+                }
+            }
+            if let Some(file) = database_config.password_file.as_ref() {
+                if url.password().is_none() {
+                    let file = File::open(file).expect("could not open the database password file");
+                    let password = BufReader::new(file)
```

Comment:
Looking back at my original issue and then at this, personally I think the right answer for password files (generally speaking) is that the entire contents of the file should be copied as-is to be the password, rather than expecting that the password can be safely trimmed or separated by a newline.

Reply:
Would the following do the trick, then ?
```rust
let password = std::fs::read_to_string(file);
```

Reply:
That seems good to me